docs/tutorials/tut-design.md

```diff
@@ -0,0 +1,369 @@
+
+There are cases where you want to build commands that perform generic well-known actions that are common across several use cases, for example Reputation commands (i.e. commands that return enrichment and reputation information about indicators, such as IPs). For those scenarios, we have standardized ways to define command names, inputs and outputs that will make interoperability much easier. For more information, check out the docs about [Generic Commands](../integrations/generic-commands-about) and [DBotScore](../integrations/dbot).
+
+#### Command Arguments
+
+Arguments are the inputs of your integration commands. They can be mandatory or not, and can have default and predefined values. More details [here](../integrations/yaml-file#commands).
+
+Command arguments should be named using [snake_case](https://medium.com/better-programming/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841).
+
+When you design your commands and their inputs, you should keep in mind how they are invoked: if a user, either manually or through a playbook, has to provide an input, where do they take that input data from? Something they know because it's obvious? Or the output of another command?
+
+The general design rule is to avoid having the SOC analyst waste time and focus by switching across multiple consoles and retrieve data from many different places. If they need to provide an input value in an XSOAR command, there should be a way to get that information within XSOAR.
+
+Let's clarify the concept with an example: imagine that you are designing a command that modifies an existing Firewall policy. For simplicity let's assume you have only two arguments: the id of the policy and the action (allow or deny). The latter argument is obvious: depending on what the user wants to do, they will set the value to `allow` or `deny` (you will use predefined values so the user can only choose between those). But what about the id of the policy? It's probably not something that they know right away: maybe they know the policy name, but you don't want them to switch context and log to a different console to find the id that corresponds to the name. So in this case, you should design your integration to make sure that there is another command that returns the list of all policies and shows their ids, or allows the user to retrieve the id from the name. This way they don't have to switch consoles.
+
+#### Command Outputs
+
+Every automation script and integration command returns several types of outputs:
+ - Human Readable: this is shown to the user in the War Room and is typically formatted in a way that is understandable by the SOC analyst. The Human Readable data is usually a subset of the entire information returned by your command.

```

Comment:
For Human Readable I would also add that it is a good practice to show the details the user see in the original product. So if in product Incident has 5 fields, then in Human Readable make sure to show these fields first